src/db/oracledb/jobs-dao.js

```diff
@@ -56,4 +120,269 @@ const getJobByJobId = async (internalId, jobId) => {
   }
 };
 
-export { getJobs, getJobByJobId };
+/**
+ * Flattens body object into a 1 layer deep object using _ as a key delimiter
+ *
+ * @param {object} body Request body
+ * @returns {object} flattened body
+ */
+const flattenBody = (body) => flatten(body, { delimiter: '_' });
+
+/**
+ * Multiple labor distributions can be created/updated at one time by concatenating the fields
+ * using '|' as the delimiter
+ *
+ * @param {object} body Request body
+ * @param {object} binds query binds to be passed in with connection.execute()
+ */
+const formatLaborDistributionForDb = (body, binds) => {
+  const laborFields = [
+    { bind: 'laborEffectiveDates', attribute: 'effectiveDate' },
+    { bind: 'laborAccountIndexCodes', attribute: 'accountIndex' },
+    { bind: 'laborFundCodes', attribute: 'fund' },
+    { bind: 'laborOrganizationCodes', attribute: 'organization' },
+    { bind: 'laborAccountCodes', attribute: 'account' },
+    { bind: 'laborProgramCodes', attribute: 'program' },
+    { bind: 'laborActivityCodes', attribute: 'activity' },
+    { bind: 'laborLocationCodes', attribute: 'location' },
+    { bind: 'laborDistributionPercentages', attribute: 'distributionPercent' },
+  ];
+
+  binds.laborCount = body.laborDistribution.length;
+  _.forEach(body.laborDistribution, (laborDist) => {
+    _.forOwn(laborFields, ({ bind, attribute }) => {
+      laborDist.effectiveDate = moment(laborDist.effectiveDate, 'YYYY-MM-DD').format('DD-MMM-YY');
+      binds[bind] = `${binds[bind] || ''}${laborDist[attribute] || ''}|`;
+    });
+  });
+};
+
+/**
+ * Transforms body into binds ready to be used with a sql execution
+ *
+ * @param {string} osuId OSU ID of a person
+ * @param {object} body Request body
+ * @param {string[]} additionalFields Extra fields to add to binds that aren't normally included
+ * @returns {object} sql binds to be used with a sql execution
+ */
+const standardBinds = (osuId, body, additionalFields) => {
+  const binds = {
+    osuId,
+    result: { type: oracledb.DB_TYPE_VARCHAR, dir: oracledb.BIND_OUT },
+  };
+
+  if (_.includes(additionalFields, 'laborDistribution') && body.laborDistribution !== undefined) {
+    formatLaborDistributionForDb(body, binds);
+  }
+
+  const flattenedBody = flattenBody(body);
+
+  // Some boolean fields need to be massaged to be compatible with db
+  if (_.includes(additionalFields, 'accruesLeaveInd')
```

Comment:
Probably better to follow the same multi-if statements style as we do in Python: https://github.com/osu-mist/style-guides/blob/master/Python.md#multiline-if-statements